lib/ansible/modules/monitoring/grafana_folder.py

```diff
@@ -0,0 +1,331 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Martin Wang (@martinwangjian)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+ANSIBLE_METADATA = {
+    'status': ['preview'],
+    'supported_by': 'community',
+    'metadata_version': '1.1'
+}
+
+DOCUMENTATION = '''
+---
+module: grafana_folder
+author:
+  - Martin Wang (@martinwangjian)
+version_added: "2.8"
+short_description: Manage Grafana folder
+description:
+  - Create, update, delete Grafana folder via API.
+options:
+  url:
+    description:
+      - The Grafana URL.
+    required: true
+    aliases: [ grafana_url ]
+  url_username:
+    description:
+      - The Grafana API user.
+    default: admin
+    aliases: [ grafana_user ]
+  url_password:
+    description:
+      - The Grafana API password.
+    default: admin
+    aliases: [ grafana_password ]
+  grafana_api_key:
+    description:
+      - The Grafana API key.
+      - If set, I(grafana_user) and I(grafana_password) will be ignored.
+  org_id:
+    description:
+      - The Grafana Organisation ID where the folder will be imported / exported.
+      - Not used when I(grafana_api_key) is set, because the grafana_api_key only belongs to one organisation..
+    default: 1
+  state:
+    description:
+      - State of the folder.
+    required: true
+    choices: [ absent, present ]
+    default: present
+  uid:
+    required: true
```

Comment:
according to the grafana API documentation, the uid is optional.

Reply:
thx, fixed